packages/components/src/Gallery/GalleryTypes.d.ts

```diff
@@ -0,0 +1,70 @@
+import { IconNames } from "../Icon";
+
+export type Sizes = "small" | "base" | "large" | "extraLarge";
+
+export type FileIconsNames = Extract<
+  IconNames,
+  "file" | "excel" | "pdf" | "video" | "word"
+>;
+
+export interface GalleryProps {
+  /**
+   * The size of the Gallery and it's files
+   * @default "base"
+   */
+  size?: "base" | "large";
+
+  /**
+   * The files for the Gallery to display
+   */
+  files: File[];
+
+  /**
+   * The max number of thumbnails before no more thumbnails are displayed
+   * unless the user clicks an action to display the rest
+   */
+  max?: number;
+
+  /**
+   * onDelete callback - this function will be called when the delete action is triggered on a Gallery image
+   */
+  onDelete?(file: File): void;
+}
+
+export interface File {
```

Comment:
Is this almost the same type as the `FileUpload`?

Reply:
Very similar yes, but `src` is a raw string instead of a promise. I tried using type extension, but I couldn't get that to work.

Reply:
you can try this

```
interface File extends Pick<FileUpload, "all" | "the" | "common" | "props"> {
  readonly src: string;
}
```

if we decide that it's supposed to be string and not a promise

Reply:
Oh yeah, inheritance is much better! Made the change 😎